src/util/ConfigManager/ConfigManager.cpp

```diff
@@ -34,24 +34,70 @@
 #include "util/json.h"
 
 namespace ad_utility {
-static auto configurationOptionsImpl(auto& configurationOptions) {
-  return std::views::transform(configurationOptions, [](auto& pair) {
-    // Make sure, that there is no null pointer.
-    AD_CORRECTNESS_CHECK(pair.second);
+// ____________________________________________________________________________
+template <typename ReturnPointer>
+std::vector<std::pair<std::string, ReturnPointer>>
+ConfigManager::configurationOptionsImpl(auto& configurationOptions,
+                                        std::string_view pathPrefix) {
+  std::vector<std::pair<std::string, ReturnPointer>> collectedOptions;
 
-    // Return a dereferenced reference.
-    return std::tie(pair.first, *pair.second);
-  });
+  std::ranges::for_each(
+      configurationOptions,
+      [&collectedOptions, &pathPrefix](auto pair) {
+        const std::string& pathToCurrentEntry =
+            absl::StrCat(pathPrefix, std::get<0>(pair));
+
+        std::visit(
+            [&collectedOptions, &pathToCurrentEntry]<typename T>(T& var) {
+              // A normal `ConfigOption` can be directly added. For a
+              // `ConfigManager` we have to recursively collect the options.
+              if constexpr (std::is_same_v<std::decay_t<T>, ConfigOption>) {
```

Comment:
We have `ad_utility::isSimilar` or `ad_utility::SimilarTo` for the very common pattern `is_same_v<decay_t<....>>`.
That makes this a little shorter.